core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp

```diff
@@ -268,18 +265,29 @@ struct CudaParallelLaunch<
       (void)prefer_shmem;
 #endif
 
-      // Copy functor to constant memory on the device
-      cudaMemcpyToSymbolAsync(kokkos_impl_cuda_constant_memory_buffer, &driver,
+      KOKKOS_ENSURE_CUDA_LOCK_ARRAYS_ON_DEVICE();
+
+      // Wait until the previous kernel that uses the constant buffer is done
+      CUDA_SAFE_CALL(cudaEventSynchronize(cuda_instance->constantMemReusable));
+
+      // Copy functor (synchronously) to staging buffer in pinned host memory
+      unsigned long* staging = cuda_instance->constantMemHostStaging;
+      memcpy(staging, &driver, sizeof(DriverType));
```

Comment:
Why do we use of plain `memcpy` instead of `cudaMemcpy` with `cudaMemcpyHostToHost`?

Reply:
So while I have to admit I had forgotten `cudaMemcpyHostToHost` even existed, what benefit do you think
would be obtained by switching to that?  We've already performed all the necessary synchronization, and
I've got to believe a compiler-understood `memcpy` is going to be lower overhead than asking CUDA to do
the host to host copy.

Reply:
It was a genuine question, I was hoping you would tell me :)